NDTensors/src/tensoroperations/generic_tensor_operations.jl

```diff
@@ -83,12 +83,15 @@ function can_contract(tensor1::Tensor, tensor2::Tensor)
   return can_contract(typeof(tensor1), typeof(tensor2))
 end
 
+using NDTensors.Expose: Exposed, expose, unexpose
 # Version where output labels aren't supplied
 @traitfn function contract(
   tensor1::TensorT1, labels_tensor1, tensor2::TensorT2, labels_tensor2
 ) where {TensorT1<:Tensor,TensorT2<:Tensor;CanContract{TensorT1,TensorT2}}
   labelsoutput_tensor = contract_labels(labels_tensor1, labels_tensor2)
-  return contract(tensor1, labels_tensor1, tensor2, labels_tensor2, labelsoutput_tensor)
+  return contract(
+    expose(tensor1), labels_tensor1, expose(tensor2), labels_tensor2, labelsoutput_tensor
```

Comment:
@mtfishman so here I am using expose to capture when the deep `datatype` is a cuarray and cutensor is loaded. When this happens the code calls cutensor based contract backend and skips the rest of the canonical NDTensors logic.

Reply:
I am reverting this and removing the function below and moving the `CuTensor` portion to `contract!`